src/java.base/unix/native/libnio/fs/UnixCopyFile.c

```diff
@@ -95,6 +103,8 @@ Java_sun_nio_fs_UnixCopyFile_transfer
             return;
         }
     } while (bytes_sent > 0);
+
+    transfer(env, dst, src, cancel);
```

Comment:
An alternative might be to call the transfer method at L95 so that it's
if (errno == EINVAL) {
    transfer(env, dst, src, cancel);
} else {
    throwUnixExceptin(env, errno);
 }
return

I suspect it might be clearer.

Reply:
I agree that is clearer.